test/e2e/storage/testsuites/provisioning.go

```diff
@@ -0,0 +1,367 @@
+	Provisioner        string
+	Parameters         map[string]string
+	DelayBinding       bool
+	ClaimSize          string
+	ExpectedSize       string
+	PvCheck            func(volume *v1.PersistentVolume) error
+	NodeName           string
+	SkipWriteReadCheck bool
+	VolumeMode         *v1.PersistentVolumeMode
+}
+
+type provisioningTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &provisioningTestSuite{}
+
+// InitProvisioningTestSuite returns provisioningTestSuite that implements TestSuite interface
+func InitProvisioningTestSuite() TestSuite {
+	return &provisioningTestSuite{
+		tsInfo: TestSuiteInfo{
+			name: "provisioning",
+			testPatterns: []testpatterns.TestPattern{
+				testpatterns.DefaultFsDynamicPV,
+			},
+		},
+	}
+}
+
+func (p *provisioningTestSuite) getTestSuiteInfo() TestSuiteInfo {
+	return p.tsInfo
+}
+
+func (p *provisioningTestSuite) skipUnsupportedTest(pattern testpatterns.TestPattern, driver drivers.TestDriver) {
+}
+
+func createProvisioningTestInput(driver drivers.TestDriver, pattern testpatterns.TestPattern) (provisioningTestResource, provisioningTestInput) {
+	// Setup test resource for driver and testpattern
+	resource := provisioningTestResource{}
+	resource.setupResource(driver, pattern)
+
+	input := provisioningTestInput{
+		testCase: StorageClassTest{
+			ClaimSize:    resource.claimSize,
+			ExpectedSize: resource.claimSize,
+		},
+		cs:    driver.GetDriverInfo().Framework.ClientSet,
+		pvc:   resource.pvc,
+		sc:    resource.sc,
+		dInfo: driver.GetDriverInfo(),
+	}
+
+	if driver.GetDriverInfo().Config.ClientNodeName != "" {
+		input.testCase.NodeName = driver.GetDriverInfo().Config.ClientNodeName
+	}
+
+	return resource, input
+}
+
+func (p *provisioningTestSuite) execTest(driver drivers.TestDriver, pattern testpatterns.TestPattern) {
+	Context(getTestNameStr(p, pattern), func() {
+		var (
+			resource     provisioningTestResource
+			input        provisioningTestInput
+			needsCleanup bool
+		)
+
+		BeforeEach(func() {
+			needsCleanup = false
+			// Skip unsupported tests to avoid unnecessary resource initialization
+			skipUnsupportedTest(p, driver, pattern)
+			needsCleanup = true
+
+			// Create test input
+			resource, input = createProvisioningTestInput(driver, pattern)
+		})
+
+		AfterEach(func() {
+			if needsCleanup {
+				resource.cleanupResource(driver, pattern)
+			}
+		})
+
+		// Ginkgo's "Global Shared Behaviors" require arguments for a shared function
+		// to be a single struct and to be passed as a pointer.
+		// Please see https://onsi.github.io/ginkgo/#global-shared-behaviors for details.
+		testProvisioning(&input)
+	})
+}
+
+type provisioningTestResource struct {
+	driver drivers.TestDriver
+
+	claimSize string
+	sc        *storage.StorageClass
+	pvc       *v1.PersistentVolumeClaim
+}
+
+var _ TestResource = &provisioningTestResource{}
+
+func (p *provisioningTestResource) setupResource(driver drivers.TestDriver, pattern testpatterns.TestPattern) {
```

Comment:
just to confirm, this no longer needs to set up generic resource? why is that?

Reply:
@davidz627

```genericVolumeTestResource```'s ```setupResource``` will create storage class and pvc, which are the scope of provisioning test. To avoid ```setupResource``` doing too much and make test do it, ```provisioningTestResource```'s ```setupResource``` is defined here and it just creates storage class object.